src/lib.rs

```diff
@@ -584,15 +588,142 @@ pub trait Context<T, E>: context::private::Sealed {
         F: FnOnce() -> C;
 }
 
+static HOOK: OnceCell<ErrorHook> = OnceCell::new();
+
+#[cfg(feature = "std")]
+pub trait ReportHandler: core::any::Any + Send + Sync {
+    fn debug(
+        &self,
+        error: &(dyn StdError + 'static),
+        f: &mut core::fmt::Formatter<'_>,
+    ) -> core::fmt::Result;
+
+    #[cfg(backtrace)]
+    fn backtrace<'a>(&'a self, error: &'a (dyn StdError + 'static)) -> &'a Backtrace;
+
+    /// Override for the `Display` format
+    fn display(
+        &self,
+        error: &(dyn StdError + 'static),
+        f: &mut core::fmt::Formatter<'_>,
+    ) -> core::fmt::Result {
+        write!(f, "{}", error)?;
+
+        if f.alternate() {
+            for cause in crate::chain::Chain::new(error).skip(1) {
+                write!(f, ": {}", cause)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(not(feature = "std"))]
+trait ReportHandler: core::any::Any + Send + Sync {
```

Comment:
I couldn't find a better way to make this trait conditionally pub, we can't export this trait under `no_std` because the `StdError` trait is private but we still need to use this trait to format the error reports via the DefaultHandler when in `no_std`.